### PR TITLE
add partitioned raft kv introduction page

### PR DESCRIPTION
**Pull request by tonyxuqqi**
Signed-off-by: tonyxuqqi <tonyxuqi@outlook.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
add partitioned-raft-kv.md

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v6.6 (TiDB 6.6 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/12978
- Other reference link(s):

### CODE REVIEW

File: partitioned-raft-kv.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+title: Partitioned Raft KV
+summary: Learn about the partitioned Raft KV feature of TiKV.
+---
+
+# Partitioned Raft KV
+
+> **Note:**
```
Comment (Oreoxmt) [inline partitioned-raft-kv.md:8]:
```suggestion
> **Warning:**
```

File: partitioned-raft-kv.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+>
+> Partitioned Raft KV is an experimental feature. It is not recommended that you use it in the production environment. This feature might be changed or removed without prior notice. If you find a bug, you can report an [issue](https://github.com/pingcap/tidb/issues) on GitHub.
+
+Before v6.6.0, TiKV's Raft-based storage engine used a single RocksDB instance to store the data of all 'Regions' of the TiKV instance.
```
Comment (Oreoxmt) [inline partitioned-raft-kv.md:12]:
```suggestion
Before v6.6.0, TiKV's Raft-based storage engine used a single RocksDB instance to store the data of all Regions of the TiKV instance.
```

File: partitioned-raft-kv.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+To support larger clusters more stably, starting from TiDB v6.6.0, a new TiKV storage engine is introduced, which uses multiple RocksDB instances to store TiKV Region data, and the data of each Region is independently stored in a separate RocksDB instance.
+
+The new engine can better control the number and level of files in the RocksDB instance, achieve physical isolation of data operations between Regions, and support stably managing more data. You can see it as TiKV managing multiple RocksDB instances through partitioning, which is why the feature is named Partitioned Raft KV.
```
Comment (Oreoxmt) [inline partitioned-raft-kv.md:16]:
```suggestion
The new engine can better control the number and level of files in each RocksDB instance, achieve physical isolation of data operations between Regions, and support stably managing more data. You can see it as TiKV managing multiple RocksDB instances through partitioning, which is why the feature is named Partitioned Raft KV.
```

File: partitioned-raft-kv.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+* A single TiKV instance needs to support more data.
+* There are many write requests.
+* Scale-in and scale-out operations are frequent.
+* The workload has a serious read and write amplification.
+* TiKV has sufficient memory.
+
+Advantages of this feature are better write performance, faster scaling speed, and larger volume of data supported with the same hardware. It can also support larger cluster scale.
+
+## Enable Partitioned Raft KV
```
Comment (Oreoxmt) [inline partitioned-raft-kv.md:30]:
```suggestion
## Usage
```

File: releases/release-6.6.0.md [hunk 5]
```diff
@@ -392,6 +394,9 @@ In v6.6.0-DMR, the key new features and improvements are as follows:
...
 | TiDB | [`tidb_stmt_summary_filename`](/tidb-configuration-file.md#tidb_stmt_summary_filename-new-in-v660) | Newly added | When statements summary persistence is enabled, this configuration specifies the file to which persistent data is written. |
 | TiKV | [`resource-control.enabled`](/tikv-configuration-file.md#resource-control) | Newly added | Whether to enable scheduling for user foreground read/write requests according to the Request Unit (RU) of the corresponding resource groups. The default value is `false`, which means to disable scheduling according to the RU of the corresponding resource groups. |
+| TiKV | [`storage.engine`](/tikv-configuration-file.md#storageengine-new-in-v660) | Newly added | This configuration item specifies the engine type. Value options are `raft-kv` and `partitioned-raft-kv`. This configuration item can only be specified when creating a cluster and cannot be modified once being specified. |
```
Comment (Oreoxmt) [inline releases/release-6.6.0.md:397]:
```suggestion
| TiKV | [`storage.engine`](/tikv-configuration-file.md#storageengine-new-in-v660) | Newly added | This configuration item specifies the engine type. Value options are `"raft-kv"` and `"partitioned-raft-kv"`. This configuration item can only be specified when creating a cluster and cannot be modified once being specified. |
```

File: tikv-configuration-file.md [hunk 6]
```diff
@@ -498,6 +498,18 @@ Configuration items related to the sharing of block cache among multiple RocksDB
...
+
+> **Note:**
+>
+> This feature is experimental. It is not recommended that you use it in the production environment. This feature might be changed or removed without prior notice. If you find a bug, you can report an [issue](https://github.com/pingcap/tidb/issues) on GitHub.
+
++ Specifies the engine type. This configuration can only be specified when creating a new cluster and cannot be modifies once being specified.
++ Value options:
+
+    + `raft-kv`: The default engine type in TiDB v6.6.0 and earlier versions.
+    + `partitioned-raft-kv`: A new storage engine introduced in TiDB v6.6.0.
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:511]:
```suggestion
    + `"raft-kv"`: The default engine type in TiDB v6.6.0 and earlier versions.
    + `"partitioned-raft-kv"`: A new storage engine introduced in TiDB v6.6.0.
```

File: tikv-configuration-file.md [hunk 7]
```diff
@@ -1095,6 +1107,29 @@ Configuration items related to RocksDB
...
+
+> **Note:**
+>
+> This feature is experimental. It is not recommended that you use it in the production environment. This feature might be changed or removed without prior notice. If you find a bug, you can report an [issue](https://github.com/pingcap/tidb/issues) on GitHub.
+
++ Specifies the flush strategy used when the memory usage of `memtable` of the current RocksDB reaches the threshold.
++ Default value: `false`
++ Value options:
+
+    + `false`: `memtable`with the largest data volume is flushed to SST files.
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:1120]:
```suggestion
    + `false`: `memtable` with the largest data volume is flushed to SST files.
```

File: tikv-configuration-file.md [hunk 8]
```diff
@@ -1095,6 +1107,29 @@ Configuration items related to RocksDB
...
+    + `true`: The earliest `memtable` is flushed to SST files. This strategy can clear the `memtable` of cold data, which is suitable for scenarios with obvious cold and hot data.
+
+### rocksdb.write-buffer-limit <span class="version-mark">New in v6.6.0</span>
+
+> **Note:**
+>
+> This feature is experimental. It is not recommended that you use it in the production environment. This feature might be changed or removed without prior notice. If you find a bug, you can report an [issue](https://github.com/pingcap/tidb/issues) on GitHub.
+
++ Specifies the total memory limit of `memtable` for all RocksDB instances in a single TiKV. The default value is 25% of the memory of the machine. It is recommended to configure a memory of at least 5 GiB. This configuration only takes effect for `partitioned-raft-kv`.
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:1129]:
```suggestion
+ Specifies the total memory limit of `memtable` for all RocksDB instances in a single TiKV. The default value is 25% of the memory of the machine. It is recommended to configure a memory of at least 5 GiB. This configuration only takes effect for Partitioned Raft KV (storage.engine="partitioned-raft-kv").
```

**Review by Oreoxmt — COMMENTED** [review]
Rest LGTM

**Review by shichun-0415 — APPROVED** [review]
LGTM

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: partitioned-raft-kv.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+To enable Partitioned Raft KV, set the configuration item [`storage.engine`](/tikv-configuration-file.md#engine-new-in-v660) to `"partitioned-raft-kv"` when creating a cluster. At the same time, you can use the configuration items [`rocksdb.write-buffer-flush-oldest-first`](/tikv-configuration-file.md#write-buffer-flush-oldest-first-new-in-v660) and [`rocksdb.write-buffer-limit`](/tikv-configuration-file.md#write-buffer-limit-new-in-v660) to control the memory usage of RocksDB when using Raft KV.
+
+## Restrictions
+
+Partitioned Raft KV has the following restrictions:
+
+* It does not support data import, replication, and backup tools, such as TiDB Lightning, TiCDC, BR, and Dumping.
+* It does not support the tikv-ctl command-line tool.
+* It cannot be used together with TiFlash.
+* You can only enable Partitioned Raft KV when creating a cluster and cannot enable it after the cluster is created.
```
Comment (benmaoer) [inline partitioned-raft-kv.md:41]:
```suggestion
* You can only enable Partitioned Raft KV when creating a cluster and cannot change the type of engine after the cluster is created.
```

**Review by benmaoer — APPROVED** [review]
LGTM

**Pull request merged**